contentcuration/contentcuration/static/js/edit_channel/admin/views.js

```diff
@@ -416,17 +416,16 @@ var ChannelItem = BaseAdminItem.extend({
             model:this.model,
             current_user: window.current_user,
             allow_leave: true,
-            onjoin: this.add_editor,
-            onleave: this.remove_editor
+            onjoin: this.fetch_editors,
+            onleave: this.fetch_editors
         });
     },
-    add_editor: function(editor){
-        this.model.set("editors", this.model.get("editors").concat(editor.toJSON()));
-        this.render();
-    },
-    remove_editor: function(editor){
-        this.model.set("editors", _.reject(this.model.get("editors"), function(user) { return user.id === editor.id }));
-        this.render();
+    fetch_editors: function(editor){
```

Comment:
Consolidated into one function as all that needs to happen is a fetch on the current set of editors